hw/xbox/nv2a/nv2a_psh.c

```diff
@@ -295,24 +296,24 @@ static QString* get_input_var(struct PixelShader *ps, struct InputInfo in, bool
 {
     QString *reg = get_var(ps, in.reg, false);
 
-    if (strcmp(qstring_get_str(reg), "0.0") != 0
-        && (in.reg != PS_REGISTER_EF_PROD
-            || strstr(qstring_get_str(reg), ".a") == NULL)) {
-        switch (in.chan) {
-        case PS_CHANNEL_RGB:
-            if (is_alpha) {
-                qstring_append(reg, ".b");
-            } else {
-                qstring_append(reg, ".rgb");
-            }
-            break;
-        case PS_CHANNEL_ALPHA:
+    switch (in.chan) {
+    case PS_CHANNEL_RGB:
```

Comment:
nit: While you're here, for readability (the values are the same, so it makes no functional difference...), would you please add `case PS_CHANNEL_BLUE:` here to show that we are handling that with the same logic?

Reply:
I believe that would trigger a duplicated-case error? Restructuring the code to check `is_alpha` first (and having 2 switch-statements) is kind of dumb, because I want to separate the `PS_CHANNEL_RGB` and `PS_CHANNEL_ALPHA` more strictly in the future (as done in the spec).

We should just remove `PS_CHANNEL_BLUE` in my opinion.
We don't have `PS_CHANNEL_AAA` either.

It's just fact that `RGB` mapped to `A`, is using the `B` channel. I'm not even sure if adding a comment ("// Only B component is used when RGB is accessed as alpha") would add any readability.

Reply:
I'm not sure about the duplicate case (haven't tried it out yet), maybe it would cause compilation error. If that's the case, we could probably add a comment at least. The reason I mentioned documenting it here, and why we may want to keep `PS_CHANNEL_BLUE` around (*though a full NVIDIA spec rename overhaul at some point would be nice...*), is because the spec specifically mentions it:
```
     The <componentUsage> parameter may be one of RGB, ALPHA, or BLUE.

     When the <portion> parameter is RGB, a <componentUsage> parameter
     of RGB indicates that the RGB portion of the indicated register
     should be assigned to the RGB portion of the combiner input variable,
     while an ALPHA <componentUsage> parameter indicates that the
     alpha portion of the indicated register should be replicated across
     the RGB portion of the combiner input variable.

     When the <portion> parameter is ALPHA, the <componentUsage>
     parameter of ALPHA indicates that the alpha portion of the indicated
     register should be assigned to the alpha portion of the combiner
     input variable, while a BLUE <componentUsage> parameter indicates
     that the blue component of the indicated register should be assigned
     to the alpha portion of the combiner input variable.
```

Reply:
Very good comment!

I didn't remember the spec also had this.

I also noticed that my previous explanation was wrong: I want to restructure code based on `is_alpha`, not `in.channel` (`PS_CHANNEL_RGB` and `PS_CHANNEL_ALPHA`).

\- I'll change my code and split this into 2 separate switch cases or equivalent.

Reply:
Added a new commit to do this.

I did not test this change myself.